src/rhsmlib/dbus/objects/register.py

```diff
@@ -279,21 +280,33 @@ def Register(self, org, username, password, options, connection_options, locale)
 
         Note this method is registration ONLY.  Auto-attach is a separate process.
         """
-        if self.is_registered():
-            raise dbus.DBusException("This system is already registered")
-
         org = dbus_utils.dbus_to_python(org, expected_type=str)
         connection_options = dbus_utils.dbus_to_python(connection_options, expected_type=dict)
         connection_options["username"] = dbus_utils.dbus_to_python(username, expected_type=str)
         connection_options["password"] = dbus_utils.dbus_to_python(password, expected_type=str)
         options = dbus_utils.dbus_to_python(options, expected_type=dict)
         locale = dbus_utils.dbus_to_python(locale, expected_type=str)
 
+        force_registration: bool = options.get("force", False)
+        system_is_registered: bool = self.is_registered()
+        if system_is_registered and not force_registration:
+            raise dbus.DBusException("This system is already registered.")
+
         with DBusSender() as dbus_sender:
             dbus_sender.set_cmd_line(sender=self.sender, cmd_line=self.cmd_line)
             Locale.set(locale)
             cp = self.build_uep(connection_options)
 
+            # If the system is registered and the 'force' option is specified,
+            # Unregister the system and register the system again.
+            if system_is_registered and force_registration:
+                log.info("This system is already registered, attempting to re-register...")
+                try:
+                    UnregisterService(cp).unregister()
+                except Exception as err:
+                    raise dbus.DBusException(str(err))
```

Comment:
most likely this is not needed, exceptions are already turned into `DBusException` (see the `dbus_handle_exceptions` decorator)